consensus/hotstuff/voteaggregator/vote_aggregator_v2.go

```diff
@@ -63,48 +65,43 @@ func NewVoteAggregatorV2(
 		queuedVotesNotifier: engine.NewNotifier(),
 	}
 
-	return aggregator, nil
-}
+	componentBuilder := component.NewComponentManagerBuilder()
 
-// Ready returns a ready channel that is closed once the engine has fully
-// started. For the propagation engine, we consider the engine up and running
-// upon initialization.
-func (va *VoteAggregatorV2) Ready() <-chan struct{} {
-	va.lm.OnStart(func() {
-		// launch as many workers as we need
-		for i := 0; i < defaultVoteAggregatorWorkers; i++ {
-			va.unit.Launch(va.queuedVotesProcessingLoop)
-		}
-	})
-	return va.lm.Started()
-}
+	// launch as many workers as we need
+	for i := 0; i < defaultVoteAggregatorWorkers; i++ {
+		componentBuilder.AddWorker(func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+			ready()
+			err := aggregator.queuedVotesProcessingLoop(ctx)
+			if err != nil {
+				ctx.Throw(err)
+			}
+		})
+	}
+
+	aggregator.ComponentManager = componentBuilder.Build()
 
-func (va *VoteAggregatorV2) Done() <-chan struct{} {
-	va.lm.OnStop(func() {
-		<-va.unit.Done()
-	})
-	return va.lm.Stopped()
+	return aggregator, nil
 }
 
-func (va *VoteAggregatorV2) queuedVotesProcessingLoop() {
+func (va *VoteAggregatorV2) queuedVotesProcessingLoop(ctx context.Context) error {
 	notifier := va.queuedVotesNotifier.Channel()
 	for {
 		select {
-		case <-va.unit.Quit():
-			return
+		case <-ctx.Done():
+			return nil
 		case <-notifier:
-			err := va.processQueuedVoteEvents()
+			err := va.processQueuedVoteEvents(ctx)
 			if err != nil {
-				va.log.Fatal().Err(err).Msg("internal error processing queued vote events")
+				return fmt.Errorf("internal error processing queued vote events: %w", err)
 			}
 		}
 	}
 }
 
-func (va *VoteAggregatorV2) processQueuedVoteEvents() error {
+func (va *VoteAggregatorV2) processQueuedVoteEvents(ctx context.Context) error {
 	for {
 		select {
-		case <-va.unit.Quit():
+		case <-ctx.Done():
```

Comment:
### :warning: 
same here: we need to listen to  `SignalerContext.ShutdownSignal()`


```golang
func (va *VoteAggregatorV2) processQueuedVoteEvents(ctx irrecoverable.SignalerContext) error {
	for {
		select {
		case <-ctx.ShutdownSignal():
```